### DISCUSSION TITLE
Orchard Core scalability in multi instances scenario

### DISCUSSION THREAD

**Discussion by MarGraz** [OrchardCMS/OrchardCore#15149] [Q&A]
Hi Everyone,

I'm new to Orchard Core and have started the learning path on Orchard Dojo (including both videos and modules). I have a newbie question: is Orchard Core's backoffice horizontally scalable?

We have a web app that will be hosted on Azure, and we're considering using Orchard Core. We need to be able to scale instances. Can Orchard Core scale instances using the same database? Is the database accessed asynchronously?

Thank you

**Comment by hishamco** [comment]
I think @Piedone is the guy because Lombiq has a great platform called "DotNest" that running on Azure

**Comment by Piedone** [comment]
The short answer is yes, but you need to configure more than what comes out of the box. What comes to my mind:

- Store configuration [in external provider](https://docs.orchardcore.net/en/latest/docs/reference/core/Configuration/) and not appsettings files.
- Store shell settings [in the database](https://docs.orchardcore.net/en/latest/docs/reference/core/Shells/#database-shells-configuration-provider).
- Store Media files [in Azure Blob](https://docs.orchardcore.net/en/latest/docs/reference/modules/Media.Azure/), [AWS S3](https://docs.orchardcore.net/en/latest/docs/reference/modules/Media.AmazonS3/), or otherwise shared.
- Store data protection keyrings [in Azure Blob](https://docs.orchardcore.net/en/latest/docs/reference/modules/DataProtection.Azure/) or [Key Vault](https://docs.orchardcore.net/en/latest/docs/reference/core/KeyVault.Azure/) or shared somehow else.
- Use the [Redis module](https://docs.orchardcore.net/en/latest/docs/reference/modules/Redis/) for distributed caching and signals.
- Instead of local Lucene search index files, use [Elasticsearch](https://docs.orchardcore.net/en/latest/docs/reference/modules/Elasticsearch/).

**Comment by MarGraz** [comment]
Thank you @hishamco and @Piedone for your reply.

I didn't find any reference in the documentation to configure Orchard Core for a multi instances scenario. 
@Piedone, I will follow your recommendations, and I will use the discussion to report any blocking problem. If we are able to configure a multi-instance environment, I will try to write a guide.

Thank you

**Reply by hishamco** [reply]
https://docs.orchardcore.net/en/latest/docs/glossary/#tenant

**Reply by MarGraz** [reply]
What I couldn't find is not the multi tenant documentation, but a guide or documentation for running multiple instances on the same machine (with the same tenant). 
In Azure, it's possible to scale your machine up by adding multiple instances of the same Web App. However, if the backoffice doesn't manage asynchronous calls to the database, scaling up your backoffice instances becomes impossible.

**Comment by ns8482e** [comment]
With redis you can enable sync between nodes, see #14318

**Comment by mariojsnunes** [comment]
> - Store shell settings [in the database](https://docs.orchardcore.net/en/latest/docs/reference/core/Shells/#database-shells-configuration-provider).

@Piedone does that mean Blob Storage for Shell is not supported?

**Reply by Piedone** [reply]
It is, but I'd use the DB since it's easier.

**Comment by mariojsnunes** [comment]
I had an issue where new tenants were not propagating to all instances. this was the fix:
<img width="334" height="51" alt="image" src="https://github.com/user-attachments/assets/75aaf806-5e0d-4804-b6fa-ee2cb8d58170" />

may be helpful to someone :)
